### PR TITLE
Convert "Add order note" screen to MVVM

### PR DESCRIPTION
**Pull request by hichamboushaba**
This is the first part of #2892, it refactors the "add order note" screen to MVVM.
As the PR is big enough, I'll do the "Add shipment Tracking fragment" refactoring in a second PR.

Please note that there is a small change in the flow, when adding a note, we will show a progress dialog: see p1607072347401500-slack-C6H8C3G23

To test: 
test different scenarios of adding notes, success, offline, error.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt [hunk 1]
```diff
@@ -364,7 +365,8 @@ class IssueRefundViewModel @AssistedInject constructor(
 
                         refundSummaryState.refundReason?.let { reason ->
                             if (reason.isNotBlank()) {
-                                noteRepository.createOrderNote(order.identifier, reason, false)
+                                val note = OrderNote(note = reason, isCustomerNote = false)
+                                orderDetailRepository.addOrderNote(order.identifier, order.remoteId, note)
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt:369]: This is a suspending function, which means we won't show the success refund message until the notes addition is done or fails, which is different from the previous behavior.
Though, I'm ignoring the result, to keep the same visible behavior as before.

**Review by AmandaRiu — CHANGES_REQUESTED** [review]
Nice work @hichamboushaba. The code looks good, but I noticed an issue during testing. The "Add" button will disappear when rotating the device or restoring the app from background while the "add note" screen is open. Typing doesn't make it visible so the only fix is to cancel out of the note and start over.

**Comment by hichamboushaba** [comment]
Thanks @AmandaRiu for the test, I will check what's wrong.

**Comment by hichamboushaba** [comment]
@AmandaRiu it appears this bug is not related to my changes, as I can reproduce on the latest beta as well, I created an issue to track it #3296, I tried investigating it a bit, but couldn't figure out what's going on yet, my guess is that it's related to the toolbar changes.
I added the commit 78c3130d751b5ced86f31f3ab1cb758116063331 to make sure that the state of the add button is restored, so I'm pretty sure when we fix the other but, we won't have any issue with my changes.

I also merged the last develop changes to use the new BaseFragment constructor, and converted the binding to a local parameter: d1ce930668c192dc637f8f64dcfd6fd24a73aa6a

**Review by AmandaRiu — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**